contracts/YearnV2Strat.sol

```diff
@@ -0,0 +1,145 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity 0.7.3;
+
+import "./IStrat.sol";
+import "./IYToken.sol";
+import "./IVault.sol";
+import "./Timelock.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "@openzeppelin/contracts/math/Math.sol";
+
+contract YTokenStrat is IStrat {
+    
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+    IVault public vault;
+    IYToken public yToken;
+    IERC20 public underlying;
+    Timelock public timelock;
+    bool public depositsPaused; // pausing only applies to Yearn deposits. Not strategy
+    uint public withdrawalCap; // 0 implies no cap
+    uint public unutilized; // similar to Uniswap reserves. Discrepancies are dangerous
+    string public name = "Yearn V2"; // for display purposes only
+
+    modifier onlyVault {
+        require(msg.sender == address(vault));
+        _;
+    }
+
+    modifier onlyOwner {
+        require(msg.sender == vault.owner()); // vault owner is strat owner
+        _;
+    }
+
+    constructor(IVault vault_, IYToken yToken_) {
+        vault = vault_;
+        yToken = yToken_;
+        timelock = Timelock(vault.timelock()); // use the same timelock from the vault
+        underlying = IERC20(yToken_.token());
+        underlying.safeApprove(address(yToken), uint(-1)); // intentional underflow
+    }
+
+    function invest() external override onlyVault {
+        uint amount = underlying.balanceOf(address(this)).sub(unutilized); // unutilized reserves do not count for incoming investment
```

Comment:
I dont like unutilized and think it is not needed. Your current implementation will cause problems imo. unutilized will just keep growing and will not be used.

Why do you use it at all? I don't see why it is needed

Reply:
Because the invest function receives an amount from the vault before getting called. Without internal accounting we cannot know how much was actually sent from the vault.

Reply:
The only other alternative is to create another contract that would hold all unutilized Dai and give an infinite approval to this contract which would keep this contract's Dai balance always empty.

Reply:
why does that matter? surely you just want to invest all you can when invest is called? you don't care how much was sent